### PR TITLE
migrate enviromentes !

### PR DESCRIPTION
**Pull request by BRAYANP**
migrate enviromentes to other file And also fix any issues with validation of the enviroments  And  add color in the log ..  More presentable !

### CODE REVIEW

File: bin/matrix-set.js [hunk 1]
```diff
@@ -18,45 +19,12 @@ Matrix.localization.init(Matrix.localesFolder, Matrix.config.locale, function ()
...
-    }
-  };
+  var environments = require('../config/environments.js');
 
   if (Matrix.pkgs.indexOf('env') === 0) {
 
     var value = Matrix.pkgs[1];
-
-    if (value && value.match(/sandbox|dev|stage|local|production|rc|hardcode/)) {
+    if (!_.isUndefined(value) && value === 'local' || value === 'local2' || value === 'dev' || value === 'rc' || value === 'stage' || value === 'production' || value === 'hardcode') {
```
Comment (eighteyes) [inline bin/matrix-set.js]: This isn't going to evaluate the way you expect, plus it's not dynamic if we add more environments.

Please switch it out for `!_.isUndefined(value) && _.keys(environments).indexOf(value)`

Comment (BRAYANP) [inline bin/matrix-set.js]: Thanks Sean ! 
Look for something like this for a long time and then I decided to do it the old :-1:

Comment (BRAYANP) [inline bin/matrix-set.js]: Test your Method of validation and It's not what we expect!

Comment (BRAYANP) [inline bin/matrix-set.js]: Done ! Issue fixed ! :+1:

Comment (eighteyes) [inline bin/matrix-set.js]: Sorry I meant `_.keys(environments).indexOf(value) !== -1`

**Pull request merged**